python/tvm/relax/op/gradient.py

```diff
@@ -0,0 +1,145 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+"""Gradient definitions for Relax operators"""
+from __future__ import annotations
+
+from tvm.relax import const
+from tvm.relay.op import register_gradient
+import tvm.relax.op.nn as nn
+from tvm.relax.op import (
+	sum,
+	less,
+	where,
+	collapse_sum_to,
+	log,
+	multiply,
+	divide,
+	negative,
+	subtract,
+	transpose,
+	sigmoid,
+	tanh,
+	ones,
+	zeros
+)
+
+
+@register_gradient("relax.add")
+def add_grad(orig, grad):
+	"""Returns [grad, grad]."""
+	return [collapse_sum_to(grad, orig.args[0].shape), collapse_sum_to(grad, orig.args[1].shape)]
+
+
+@register_gradient("relax.subtract")
+def subtract_grad(orig, grad):
+	"""Returns [grad, -grad]."""
+	return [collapse_sum_to(grad, orig.args[0].shape), collapse_sum_to(negative(grad), orig.args[1].shape)]
+
+
+@register_gradient("relax.multiply")
+def multiply_grad(orig, grad):
+	"""Returns [grad * y, grad * x]."""
+	x, y = orig.args
+	return [collapse_sum_to(multiply(grad, y), x.shape), collapse_sum_to(multiply(grad, x), y.shape)]
+
+
+@register_gradient("relax.transpose")
+def transpose_grad(orig, grad):
+	"""Returns grad transposed over the complement of original transpose axes"""
+	"""TODO: Do not support more than one dimensions"""
```

Comment:
I’m curious about this “to do”. Do you mean this gradient operator only supports the case of transposing a 2-dim tensor? This transpose operator will reverse the axis order by default if we don’t specify the `axes` field.